src/components/structures/RoomView.js

```diff
@@ -111,27 +116,25 @@ module.exports = React.createClass({
         // We can /peek though. If it fails then we present the join UI. If it
         // succeeds then great, show the preview (but we still may be able to /join!).
         if (!this.state.room) {
-            console.log("Attempting to peek into room %s", this.props.roomId);
-            MatrixClientPeg.get().peekInRoom(this.props.roomId).done(() => {
-                // we don't need to do anything - JS SDK will emit Room events
-                // which will update the UI. We *do* however need to know if we
-                // can join the room so we can fiddle with the UI appropriately.
```

Comment:
Both work. I had no reason to prefer one or the other. We just need to make sure that if we discover a new room (via peeking or via joining) that we check which screen to show (join UI etc)